contract/views/contract.xml

```diff
@@ -0,0 +1,653 @@
+                                attrs="{'readonly': [('is_terminated','=',True)]}"
+                            />
+                        </group>
+                        <group>
+                            <field
+                                name="contract_template_id"
+                                attrs="{'readonly': [('is_terminated','=',True)]}"
+                                domain="['|', ('contract_type', '=', contract_type), ('contract_type', '=', False)]"
+                                context="{'default_contract_type': contract_type}"
+                            />
+                            <field name="contract_type" invisible="1" required="1" />
+                            <field
+                                name="fiscal_position_id"
+                                attrs="{'readonly': [('is_terminated','=',True)]}"
+                            />
+                            <field
+                                name="journal_id"
+                                required="1"
+                                attrs="{'readonly': [('is_terminated','=',True)]}"
+                            />
+                            <field name="tag_ids" widget="many2many_tags" />
+                        </group>
+                    </group>
+                    <group name="recurring_invoices">
+                        <field name="line_recurrency" class="oe_inline" />
+                        <label for="line_recurrency" />
+                        <group attrs="{'invisible': [('line_recurrency', '=', True)]}">
+                            <label for="recurring_interval" />
+                            <div class="o_row">
+                                <field
+                                    name="recurring_interval"
+                                    attrs="{'required': [('line_recurrency', '=', False)]}"
+                                    class="oe_inline"
+                                    nolabel="1"
+                                />
+                                <field
+                                    name="recurring_rule_type"
+                                    attrs="{'required': [('line_recurrency', '=', False)]}"
+                                    class="oe_inline"
+                                    nolabel="1"
+                                />
+                            </div>
+                            <field
+                                name="recurring_invoicing_type"
+                                attrs="{'required': [('line_recurrency', '=', False)]}"
+                            />
+                        </group>
+                        <group attrs="{'invisible': [('line_recurrency', '=', True)]}">
+                            <field
+                                name="date_start"
+                                attrs="{'required': [('line_recurrency', '=', False)]}"
+                            />
+                            <field name="date_end" />
```

Comment:
- `date_end` must be editable
- `recurring_next_date` must be computed after recurrence change
![1](https://user-images.githubusercontent.com/12665409/91305850-f68bc380-e7ab-11ea-85f8-d2ff07b8289b.png)

Reply:
Fixed with latest changes that cleans a bit the definition across inheritance and make more use of computed.